vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -1740,3 +1741,565 @@ def execute_model(
             prompt_logprobs_dict={},
             pooler_output=pooler_output)
         return model_output
+
+
+class ChunkedPrefillModelRunner(ContinuousBatchingSpyreModelRunner):
+
+    def _prepare_prompt(self, _):
+        AssertionError(
+            "Should not call this method on chunked prefill implementation")
+
+    def _prepare_chunked_prefill(self, scheduler_output: SchedulerOutput):
+        '''
+        Cases / Scenarios for the chunked prefill with right padding.
+ 
+
+        X    - Padding
+        T    - Token
+        O    - Right padding / unused slot
+        |    - Blocks separator
+        ||   - Chunks separator
+        ...  - Trimmed sequence
+
+        For the following "drawing" consider blocks of 4 tokens
+        and chunks of 8 tokens
+
+        # Case I
+
+        Prompt fits in the chunk with left padding
+
+        3 tokens
+        1 block
+        1 chunk
+        4 left padding
+
+        X X X X | T T T O || O O O O | O O O O || O O O O | O O O O || ...
+
+        Variation: Prompt fits in the chunk but no left padding needed
+
+        T tokens
+        2 block
+        1 chunk
+        0 left padding
+
+        T T T T | T T T O || O O O O | O O O O || O O O O | O O O O || ...
+
+        ---
+        # Case II 
+
+        Prompt is greater than chunk, and it contains left padding
+
+        10 tokens
+        4 blocks
+        2 chunks
+        4 left padding
+
+        X X X X | T T T T || T T T T | T T O O || O O O O | O O O O || ...
+        
+        # Case III 
+
+        No left paddings and more than one chunk
+
+        13 tokens
+        4 blocks
+        2 chunks
+        0 left padding
+
+        T T T T | T T T T || T T T T | T O O O || O O O O | O O O O || ...
+
+        NOTE: The goal of this "illustration" is to depics strategies to write
+        code to create the chunks, not necessarily enumerate the possible 
+        scenario. Of course there are interpretations where these cases 
+        overlaps. 
+        
+        '''
+
+        # Get request id from new request or cached request
+        req_id = scheduler_output.scheduled_new_reqs[0].req_id if \
+            len(scheduler_output.scheduled_new_reqs) == 1 \
+                else scheduler_output.scheduled_cached_reqs.req_ids[0]
+
+        request = self.requests[req_id]
+        prompt_token_ids = request.prompt_token_ids
+
+        chunk_size = self.scheduler_config.max_num_batched_tokens
+        chunk_blocks = chunk_size // self.block_size
+        num_computed_tokens = request.num_computed_tokens
+        # basic check, if the computed tokens is divisible by the
+        # chunk size
+        assert num_computed_tokens % chunk_size == 0
```

Comment:
This is checking that the number of previously computed tokens is divisible by the chunk size, right?

If so I don't think that's true- when we do left padding with full blocks for the first chunk then we will compute a number of tokens smaller than the chunk size